src/prevector.h

```diff
@@ -35,6 +35,8 @@
  */
 template<unsigned int N, typename T, typename Size = uint32_t, typename Diff = int32_t>
 class prevector {
+    static_assert(std::is_trivially_copyable_v<T>);
```

Comment:
Out of interest, I am wondering to what extend trivially copyable is related to trivially destructible?

Reply:
Per the standard a class is only trivially copyable when it also has a trivial destructor: https://eel.is/c++draft/class#prop-1.3